### PR TITLE
HSDO-186: Added a field collection throttle

### PR DESCRIPTION
**Pull request by sherakama**
Provides a way to throttle how many freaking field collections get created.

### CODE REVIEW

**Comment by jbickar** [comment]
Oooh, I think I like this idea a lot.

File: stanford_courses.module [hunk 1]
```diff
@@ -127,15 +127,27 @@ function stanford_courses_feeds_set_target($source, &$entity, $target, $value) {
...
     feeds_alter('feeds_processor_targets', $sub_targets[$target], 'field_collection_item', $target);
   }
+
   $_sub_targets = $sub_targets[$target];
 
   $value = is_array($value) ? $value : array($value);
   $info = field_info_field($target);
 
+  // Sometimes courses have waaaaaaaay too many sections and we might want to
+  // restrict how many are being set because it breaks everyting. This is how.
```
Comment (sherakama) [inline stanford_courses.module]: Yes, EVERY TING!

Comment (jbickar) [inline stanford_courses.module]: How can you argue with a man who has cruise control for awesome?

**Comment by jbickar** [comment]
_Quid pro quo_: This will get tested and merged tomorrow if you merge https://github.com/SU-SWS/stanford_s3fs_if/pull/1/files

**Comment by jbickar** [comment]
Well played.

What are the steps to test? Our resource constraints aren't on local, correct?

So I need to do:

1. Create a course importer on a site on Anchorage that includes ECON1
2. Import
3. Watch it puke
4. Check out this branch
5. `drush vset stanford_courses_limit_field_collections 20`
6. Re-import
7. Rejoice

Correct?

I'm cool with this being limited to `drush vset` for now; I like this approach for phased implementation. GUI can come later.

**Comment by sherakama** [comment]
It will probably puke on the sites environment as well. So you can take your pick of either.

The importer that INTNLREL was using is:
http://explorecourses.stanford.edu/search?q=IR%3A%3A*&view=xml-20140630&page=0&academicYear=&filter-coursestatus-Active=on&filter-coursestatus-I=on&collapse=%2C3%2C9%2C&filter-component-LEC=on&filter-component-SEM=on&filter-component-DIS=on&filter-component-LAB=on&filter-component-LBS=on&filter-component-ACT=on&filter-component-CAS=on&filter-component-COL=on&filter-component-WKS=on&filter-component-IDS=on&filter-component-ISF=on&filter-component-ISS=on&filter-component-ITR=on&filter-component-API=on&filter-component-LNG=on&filter-component-PRA=on&filter-component-PRC=on&filter-component-RES=on&filter-component-SCS=on

And I used this plugin to count the FCs:
https://github.com/sherakama/countfcs

The goal of this plugin is to ensure that no more than N FCs get saved to any imported node.

**Comment by jbickar** [comment]
Interestingly, that International Relations importer didn't puke on Sites. Trying https://explorecourses.stanford.edu/search;jsessionid=1chal139qreeb1bi1eifyvn6rw?view=catalog&academicYear=&page=0&q=ECON&filter-departmentcode-ECON=on&filter-coursestatus-Active=on&filter-term-Spring=on now.

**Comment by jbickar** [comment]
This works great. It doesn't work if you have an balky importer created before you set this setting, but it works on new importers.

I really like this approach because it gives us a lightweight way to see what the impacts are of cutting off field collections at an arbitrary number.

**Comment by sherakama** [comment]
Woot. 

Yeah, that one of the side effects that we are going to have to think about. Even on the same importer if you change the value of field_collections allowed and nothing has changed on the remote end, none of the courses get updated. 

But that is for future us.

**Pull request merged**